src/utils/uint256.cairo

```diff
@@ -19,3 +33,39 @@ func uint256_exp{range_check_ptr}(a: Uint256, b: Uint256) -> Uint256 {
     let (res, _) = uint256_mul(a, pow);
     return res;
 }
+
+// @notice Performs sign extension on a Uint256 number.
+// @param x The number to be sign extended.
+// @param b The size in bytes minus one of x to consider.
+// @return The result of sign extending x from position b.
+// If b > 31, the result is x. Otherwise, the sign bit is extended from position b.
+func uint256_signextend{range_check_ptr, bitwise_ptr: BitwiseBuiltin*}(
+    x: Uint256, b: Uint256
+) -> Uint256 {
+    alloc_locals;
+    let (x_fits_in_evm_word) = uint256_lt(b, Uint256(32, 0));
+    if (x_fits_in_evm_word == 0) {
+        return x;
+    }
+
+    let (mul, _) = uint256_mul(b, Uint256(8, 0));
```

Comment:
at this point it's guaranteed that `b <= 31`, so we can drop the Uint256 and use `b.low * 8 + 7`

Reply:
given the fact that we have 32 possibilities, I think that we should use a hard coded table of values to be able to do in one line

```
let (sign, value) = unsigned_div_rem(x, mask);
```

Reply:
and then something like

```
let value = Uint256.max - bytes[s] + value
```